### PR TITLE
feat: Centering the dropping placeholder on the cursor

### PR DESCRIPTION
**Pull request by corpille**

### CODE REVIEW

File: src/components/GridLayout.vue [hunk 1]
```diff
@@ -539,6 +540,7 @@
                 if (this.dragEnterCounter === 0) {
                     this.removeDroppingPlaceholder();
                 }
+                this.$emit('drag-leave')
```
Comment (jledentu) [inline src/components/GridLayout.vue]: Is it necessary? If needed, setting an handler on native event should be enough. 

If you really want this, you have to emit the event only when `dragEnterCounter` is 0.

File: src/components/GridLayout.vue [hunk 2]
```diff
@@ -583,14 +599,15 @@
...
-                    const {x, y} = calcXY(positionParams, droppingPosition.top, droppingPosition.left, w, h);
+                    const {x, y} = calcXY(positionParams, pos.top, pos.left, w, h);
 
                     if (x !== this.droppingPlaceholder.x || y !== this.droppingPlaceholder.y) {
                         this.droppingPlaceholder.x = x;
                         this.droppingPlaceholder.y = y;
                         this.dragEvent('dragmove', DROPPING_ID, x, y, h, w);
                     }
                 }
+                this.$emit('drag-over')
```
Comment (jledentu) [inline src/components/GridLayout.vue]: Is it necessary? What's the impact on performance?

File: src/components/GridLayout.vue [hunk 3]
```diff
@@ -526,6 +526,7 @@
                 event.preventDefault();
                 event.stopPropagation();
                 this.dragEnterCounter++;
+                this.$emit('drag-enter')
```
Comment (jledentu) [inline src/components/GridLayout.vue]: Is it necessary? If needed, setting an handler on native event should be enough.

If you really want this, you have to emit the event only when `dragEnterCounter` becomes > 0.

File: src/components/GridLayout.vue [hunk 4]
```diff
@@ -570,9 +572,23 @@
                     containerWidth: this.width !== null ? this.width : 100,
                 };
 
-                if (!this.droppingPlaceholder) {
-                    const {x, y} = calcXY(positionParams, droppingPosition.top, droppingPosition.left, w, h);
+                const width = positionParams.containerWidth / positionParams.cols * (w / 2);
```
Comment (jledentu) [inline src/components/GridLayout.vue]: There is also code in `GridItem.vue` that computes the size of an item (by also taking `margin` into account). I suggest to mutualize some code by adding a function in `computeUtils.js` and using a function like this:

```js
const { width, height } = calcItemSize(positionParams, w, h);
const offset = {
  left: width / 2,
  top: height / 2,
};

 const pos = {
  top: droppingPosition.top - offset.top,
  left: droppingPosition.left - offset.left
};
```

**Review by jledentu — COMMENTED** [review]
Works nice! I made some suggestions to improve the code.

File: src/components/GridLayout.vue (hunk 1 above)
Comment (corpille) [reply src/components/GridLayout.vue]: Indeed a native event would work better.

File: src/components/GridLayout.vue (hunk 4 above)
Comment (corpille) [reply src/components/GridLayout.vue]: I've made the changes to mutualise the code in both `GridItem.vue` and `GridLayout.vue`

**Comment by jledentu** [comment]
Congrats! It works well, let's merge it.

**Pull request merged**